### PR TITLE
SWC-6177

### PR DESCRIPTION
**Pull request by yjyoo773**

### CODE REVIEW

**Comment by yjyoo773** [comment]
Screen Shot 2022-07-01 at 10 02 48 AM

File: src/lib/containers/table/datasets/DatasetItemsEditor.tsx [hunk 1]
```diff
@@ -216,10 +212,20 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
...
           ...datasetToUpdate,
           items: items,
         }
+        displayToast(SAVE_THE_DATASET_TO_CONTINUE, 'info', {
+          title: toastMessageTitle,
+          primaryButtonConfig: {
+            text: 'Save changes to Draft',
+            onClick: () => mutation.mutate(updatedDataSet),
+          },
+        })
```
Comment (nickgros) [inline src/lib/containers/table/datasets/DatasetItemsEditor.tsx:226]: I think `displayToast` should be called outside of the `setDatasetToUpdate` function. This creates the potential for the following:
* User adds item `a` to Dataset, toast is shown where `onClick` saves Dataset has item `a`
* User removes item `a` and adds item `b` to Dataset, toast is shown where `onClick` saves Dataset with  `b`
* User clicks Save button on first toast, dataset is saved with just `a`

Instead, let's move this out of the setter and have the onClick call `mutation.mutate(datasetToUpdate)`

File: src/lib/containers/table/datasets/DatasetItemsEditor.tsx [hunk 2]
```diff
@@ -231,19 +237,26 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
...
-      } removed from the Dataset`,
+      displayToast(SAVE_THE_DATASET_TO_CONTINUE, 'info', {
+        title: `${selectedIds.size} Item${
+          selectedIds.size === 1 ? '' : 's'
+        } removed from the Dataset`,
+        primaryButtonConfig: {
+          text: 'Save changes to Draft',
+          onClick: () => mutation.mutate(updatedDataSet),
+        },
+      })
```
Comment (nickgros) [inline src/lib/containers/table/datasets/DatasetItemsEditor.tsx:256]: Same scenario as above, please move out of the setter function.

**Review by nickgros — CHANGES_REQUESTED** [review]

File: src/lib/containers/table/datasets/DatasetItemsEditor.tsx (hunk 1 above)
Comment (yjyoo773) [reply src/lib/containers/table/datasets/DatasetItemsEditor.tsx:226]: The reason I added `displayToast` inside `setDatasetToUpdate` was that when setting as `mutation.mutate(datasetToUpdate)` the state isn't updated when the button is clicked. Also, the toast title depends on the data within `setDatasetToUpdate`.

**Pull request merged**